data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/SinkModel.java

```diff
@@ -86,23 +102,53 @@ private static class SinkInternalJsonModel extends InternalJsonModel {
         @JsonProperty("tags_target_key")
         private final String tagsTargetKey;
 
+
+        @JsonInclude(JsonInclude.Include.NON_EMPTY)
+        @JsonProperty("include_keys")
+        private final List<String> includeKeys;
+
+        @JsonInclude(JsonInclude.Include.NON_EMPTY)
+        @JsonProperty("exclude_keys")
+        private final List<String> excludeKeys;
+
         @JsonCreator
-        private SinkInternalJsonModel(@JsonProperty("routes") final List<String> routes, @JsonProperty("tags_target_key") final String tagsTargetKey) {
-            super();
-            this.routes = routes != null ? routes : new ArrayList<>();
-            this.tagsTargetKey = tagsTargetKey;
+        private SinkInternalJsonModel(@JsonProperty("routes") final List<String> routes, @JsonProperty("tags_target_key") final String tagsTargetKey, @JsonProperty("include_keys") final List<String> includeKeys, @JsonProperty("exclude_keys") final List<String> excludeKeys) {
+            this(routes, tagsTargetKey, includeKeys, excludeKeys, new HashMap<>());
         }
 
-        private SinkInternalJsonModel(final List<String> routes, final String tagsTargetKey, final Map<String, Object> pluginSettings) {
+        private SinkInternalJsonModel(final List<String> routes, final String tagsTargetKey, final List<String> includeKeys, final List<String> excludeKeys, final Map<String, Object> pluginSettings) {
             super(pluginSettings);
             this.routes = routes != null ? routes : new ArrayList<>();
+            this.includeKeys = includeKeys != null ? preprocessingKeys(includeKeys) : new ArrayList<>();
+            this.excludeKeys = excludeKeys != null ? preprocessingKeys(excludeKeys) : new ArrayList<>();
             this.tagsTargetKey = tagsTargetKey;
         }
+
+
+        /**
+         * Pre-processes a list of Keys and returns a sorted list
+         * The keys must start with `/` and not end with `/`
+         *
+         * @param keys a list of raw keys
+         * @return a sorted processed keys
+         */
+        private List<String> preprocessingKeys(final List<String> keys) {
+            if (keys.contains("/")) {
+                return new ArrayList<>();
+            }
+            List<String> result = keys.stream()
+                    .map(k -> k.startsWith("/") ? k : "/" + k)
```

Comment:
Do we have to require the json pointer at the start? I think just `some_key` and `/some_key` should both work

Reply:
This is one-time work during the start to unify all the json paths for check. Customers can put both in the yaml.